tools/wpt/testfiles.py

```diff
@@ -79,13 +80,21 @@ def branch_point():
     return branch_point
 
 
-def files_changed(revish, ignore_dirs=None, include_uncommitted=False, include_new=False):
-    """Get and return files changed since current branch diverged from master,
-    excluding those that are located within any directory specifed by
-    `ignore_changes`."""
-    if ignore_dirs is None:
-        ignore_dirs = []
+def compile_ignore_rule(rule):
+    rule = rule.replace(os.path.sep, "/")
+    parts = rule.split("/")
+    re_parts = []
+    for part in parts:
+        if part.endswith("**"):
```

Comment:
Parsing out what the body of this loop does isn't easy, for me. Maybe a test for this that hits each of the branches? (Don't need to apply the rules, just stringify the resulting regexes and asssert that they are what they should be.)

Reply:
Tests now in `test_files_changed_ignore_rules()`